mungegithub/mungers/lgtm_handler.go

```diff
@@ -0,0 +1,166 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mungers
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"k8s.io/contrib/mungegithub/features"
+	"k8s.io/contrib/mungegithub/github"
+	"k8s.io/contrib/mungegithub/mungers/mungerutil"
+
+	"github.com/golang/glog"
+	githubapi "github.com/google/go-github/github"
+	"github.com/spf13/cobra"
+)
+
+const (
+	lgtmRemovedBody = "PR changed after LGTM, removing LGTM."
+)
+
+// LGTMHandler will
+// - apply the LGTM label if reviewer has said so, or
+// - remove the LGTM label from an PR which has been updated since the reviewer added LGTM
+type LGTMHandler struct{}
+
+func init() {
+	l := LGTMHandler{}
+	RegisterMungerOrDie(l)
+	RegisterStaleComments(l)
+}
+
+// Name is the name usable in --pr-mungers
+func (LGTMHandler) Name() string { return "lgtm-after-commit" }
+
+// RequiredFeatures is a slice of 'features' that must be provided
+func (LGTMHandler) RequiredFeatures() []string { return []string{} }
+
+// Initialize will initialize the munger
+func (LGTMHandler) Initialize(config *github.Config, features *features.Features) error {
+	return nil
+}
+
+// EachLoop is called at the start of every munge loop
+func (LGTMHandler) EachLoop() error { return nil }
+
+// AddFlags will add any request flags to the cobra `cmd`
+func (LGTMHandler) AddFlags(cmd *cobra.Command, config *github.Config) {}
+
+// Munge is the workhorse the will actually make updates to the PR
+func (h LGTMHandler) Munge(obj *github.MungeObject) {
+	if !obj.IsPR() {
+		return
+	}
+
+	lastModified := obj.LastModifiedTime()
+	if !obj.HasLabel(lgtmLabel) {
+		allAssignees := append(obj.Issue.Assignees, obj.Issue.Assignee)
+		reviewers := mungerutil.FilterValid(allAssignees)
+		if len(reviewers) == 0 {
+			return
+		}
+		comments, err := obj.ListComments()
+		if err != nil {
+			glog.Errorf("unexpected error getting comments: %v", err)
+		}
+		users := findLGTMCommenters(comments, lastModified)
+		for _, user := range users {
+			// TODO: support more complex policies for multiple reviewers.
+			// See https://github.com/kubernetes/contrib/issues/1389#issuecomment-235161164
+			if isReviewer(user, reviewers) {
+				obj.AddLabel(lgtmLabel)
+				return
+			}
+		}
+		if len(users) != 0 {
+			// if somebody that is NOT the reviewer says "LGTM", it would be nice to mention the reviewer, rather than just ignore.
+			obj.WriteComment(fmt.Sprintf("LGTM from users (%v). Can reviewers (%v) review this?", users, reviewers))
```

Comment:
> Currently the assignee is the reviewer/owner. We should change this behavior. Can we add a TODO here and handle it in separate PR?

@bgrant0607 I mean "we probably would change this behavior". I have added a note there saying "if assignees = reviewers not true anymore..."